File: comms/push-notifications/credentials/list-credentials/list-credentials.php
<?php

namespace Example;

use Twilio\Comms\TwilioComms;
use Twilio\Comms\PushNotifications\Credentials\Requests\CredentialsListRequest;
use DateTime;
use Twilio\Comms\Types\PushCredentialType;

$client = new TwilioComms(
    accountId: 'TWILIO_ACCOUNT_SID',
    authToken: 'TWILIO_AUTH_TOKEN',
);
$client->pushNotifications->credentials->list(
    new CredentialsListRequest([
        'startDate' => new DateTime('2024-01-15T09:30:00Z'),
        'endDate' => new DateTime('2024-01-15T09:30:00Z'),
        'appName' => 'appName',
        'credentialType' => PushCredentialType::Apn->value,
        'isDefault' => true,
        'pageToken' => 'pageToken',
        'pageSize' => 50,
    ]),
);

File: comms/push-notifications/credentials/list-credentials/list-credentials.rb
require "twiliocomms"

client = TwilioComms::Client.new(
  account_id: "TWILIO_ACCOUNT_SID",
  auth_token: "TWILIO_AUTH_TOKEN"
)

client.push_notifications.credentials.list(
  start_date: "2024-01-15T09:30:00Z",
  end_date: "2024-01-15T09:30:00Z",
  app_name: "appName",
  credential_type: "APN",
  is_default: true,
  page_token: "pageToken",
  page_size: 50
)
